tools/clang/tools/dxcompiler/dxcutil.cpp

```diff
@@ -241,11 +230,13 @@ HRESULT ValidateAndAssembleToContainer(AssembleInputs &inputs) {
       debugModule.Size = pDebugModuleStream->GetPtrSize();
 
       IFT(pValidator2->ValidateWithDebug(inputs.pOutputContainerBlob,
-                                         DxcValidatorFlags_InPlaceEdit,
+                                         DxcValidatorFlags_InPlaceEdit |
+                                             inputs.ValidationFlags,
                                          &debugModule, &pValResult));
     } else {
-      IFT(pValidator->Validate(inputs.pOutputContainerBlob,
-                               DxcValidatorFlags_InPlaceEdit, &pValResult));
+      IFT(pValidator->Validate(
+          inputs.pOutputContainerBlob,
+          DxcValidatorFlags_InPlaceEdit | inputs.ValidationFlags, &pValResult));
```

Comment:
It might be slightly less intrusive if we just set the `InPlaceEdit` bit in `ValidationFlags` when `AssembleInputs` is created and could just pass in `inputs.ValidationFlags` here. I don't feel too strongly about it though if you prefer to leave it this way.

Reply:
Good idea.
Updated.